### PR TITLE
mock: re-allow running mock as root

### PR DESCRIPTION
**Pull request by praiskup**
Turns out many users are running mock directly under root.  So don't limit them; moving the non-root check into simple_load_config.

Fixes: https://bugzilla.redhat.com/2135203
Fixes: #990

### CODE REVIEW

File: mock/py/mockbuild/config.py [hunk 1]
```diff
@@ -764,6 +759,12 @@ def list_configs(config_opts):
 @traceLog()
 def simple_load_config(name, config_path=None):
     """ wrapper around load_config() intended by use 3rd party SW """
+
+    for the_id in getresuid() + getresgid():
+        if the_id != 0:
+            continue
+        raise exception.ConfigError("Can't parse Mock configuration as root")
```
Comment (xsuchy) [inline mock/py/mockbuild/config.py:766]: I would argue, that this should be just a warning. If someone wants to do that. We should allow them to shoot in their leg.

**Review by xsuchy — APPROVED** [review]
+1

**Comment by AdamWill** [comment]
:+1: . Are you planning to do a release and build today, or should I backport this? I need a build for Rawhide at least quite urgently to fix openQA tests (and I assume Fedora CI folks would like the same to fix CI tests).

**Comment by praiskup** [comment]
I'm going to release it right now.  Thank you for the review!

**Pull request closed without merging**